src/apps/backend/modules/otp/otp-service.ts

```diff
@@ -0,0 +1,43 @@
+import { PhoneNumber } from '../account';
+import { SMSService } from '../communication';
+import CommunicationUtil from '../communication/communication-util';
+
+import OtpWriter from './internal/otp-writer';
+import {
+  Otp,
+  OtpRequestError,
+} from './types';
+
+export default class OtpService {
+  public static async createOtp(
+    phoneNumber: PhoneNumber,
+  ): Promise<Otp> {
+    const isValidPhoneNumber = CommunicationUtil.checkPhoneNumberValidation(
+      phoneNumber.toString(),
+    );
+
+    if (!isValidPhoneNumber) {
+      throw new OtpRequestError('Invalid phone number');
+    }
+
+    const otp = await OtpWriter.createOtp(phoneNumber);
+
+    if (!otp) {
+      throw new OtpRequestError('Failed to create OTP');
```

Comment:
Poor copyright. Please use generative AI, google and general guidance to craft better usage facing message

Reply:
Is this one better `Failed to send OTP: Internal server error`?
or `OTP creation failed: Internal server error`